torch/_higher_order_ops/cond.py

```diff
@@ -305,7 +305,8 @@ def _has_potential_branch_input_mutation(branch, inputs):
     bit restrictive as the branch must be traceable.
     """
     try:
-        gm = make_fx(branch)(*inputs)
+        with disable_proxy_modes_tracing(pre_dispatch=True):
+            gm = make_fx(branch, pre_dispatch=True)(*inputs)
```

Comment:
Why do we need the "detect input mutation" logic in cond to use pre-dispatch tracing? (we're not returning the traced graph here, we're just determining if the function has input mutation/aliasing)

Reply:
Also, why do we now need to disable_proxy_modes_tracing (vs. we didn't need it before?)